src/FormatNumber/Parser.elm

```diff
@@ -378,3 +380,61 @@ parse locale original =
                 | prefix = locale.zeroPrefix
                 , suffix = locale.zeroSuffix
             }
+
+
+{-| Given a `Locale` parses a `String` into a `Maybe Float`
+-}
+parseString : Locale -> String -> Maybe Float
+parseString locale value =
+    let
+        isNegative : Bool
+        isNegative =
+            negativePrefixMatch && negativeSuffixMatch
+
+        negativePrefixMatch : Bool
+        negativePrefixMatch =
+            value
+                |> String.startsWith locale.negativePrefix
+
+        negativeSuffixMatch : Bool
+        negativeSuffixMatch =
+            value
+                |> String.endsWith locale.negativeSuffix
+
+        chrs : Regex.Regex
+        chrs =
+            "[^\\d]"
+                |> Regex.fromString
+                |> Maybe.withDefault Regex.never
+
+        splitValue : String -> List String
+        splitValue nbr =
+            nbr
+                |> String.split locale.decimalSeparator
+                |> List.map (Regex.replace chrs (\_ -> ""))
+
+        sumNumber : Float -> Float -> Float -> Float
+        sumNumber i d pow =
+            if isNegative then
+                -1 * (i + d / (10 ^ pow))
+
+            else
+                i + d / (10 ^ pow)
```

Comment:
This function has two responsibilities, coupling the logic too much. Let's think about a `toFloat` and we multiply by `-1` at the end of the `parseSting` chain. Also, why do we multiply by `-1` here and then, again, in line 432? This logic is confusing.

Can we rename `i` and `d`? They are not meaningful variable names. What about `integers` and `decimals`? And talking about meaningful names, can you explain in plain English what `sumNumber` does? I would like to come up with a better name for this function.

Finally, I would treat `pow` as part of the logic of this function, and not as something calculated elsewhere.

Keeping all that in mind, I think of a refactor `sumNumber` as `toFloat : (String, String) -> Float`, something among these lines:

```elm
toFloat parts =
    let
        pow : Float
        pow =
             parts
                |> Tuple.second
                |> String.length

        decimals : Float
        decimals =
             parts
                |> Tuple.second
                |> String.toFloat
                |> Maybe.withDefault 0.0

        integers : Float
        integers =
             parts
                |> Tuple.first
                |> String.toFloat
                |> Maybe.withDefault 0.0 
    in
    integers + decimals / (10 ^ pow) 
```

Reply:
Added function with name `joinParts`. It is opposite of `splitInParts`. It also performs an opposite function. It takes integer and decimal strings and returns `Maybe Float`. Rules are encoded in tests in comments.